sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -134,6 +131,32 @@ def fit(self, X, y=None):
 
         return self
 
+    def add_element(self, f, v, feature_names, vocab, fitting=True,
+                    transforming=False, indices=None, values=None):
+        if not isinstance(v, six.string_types) and isinstance(v, Iterable):
+            for vv in v:
+                self.add_element(f, vv, feature_names, vocab,
+                                 fitting, transforming, indices, values)
+        else:
+            if isinstance(v, six.string_types):
+                feature_name = "%s%s%s" % (f, self.separator, v)
+                v = 1
+            elif isinstance(v, Number) or (v is True) or\
```

Comment:
```
>>> isinstance(True, Number)
True
```

But more generally, I don't understand why we need this check. We've never done it before and it only helps in a case where creating the matrix will ultimately fail. And there might be numpy types that we should be allowing here.

Reply:
```
In [11]: isinstance(np.int32(10), numbers.Number)
Out[11]: True```